### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by roberth**
- Extend the shell
 - Add `ci.nix`

### CODE REVIEW

File: nix/ci.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    dimension
+    ;
+
+in
+  dimension "System" {
+    "x86_64-linux" = {};
+    # "x86_64-darwin" = {};
+  } (system: {}:
+
+    dimension "Nixpkgs release" {
```
Comment (domenkozar) [inline nix/ci.nix]: This is not needed, it should just be bumped.

File: nix/dimension.nix [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                myProject {
+                  inherit nixpkgs;
+                  testRunnerPkgs = releases."${latest}";
+                }
+
+              );
+        in releases;
+
+   */
+  dimension = name: m: f:
```
Comment (domenkozar) [inline nix/dimension.nix]:
```suggestion
  dimension = name: attrs: f:
```

File: nix/dimension.nix [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    lib.mapAttrs
+      (k: v:
+       let o = f k v;
+       in o // { recurseForDerivations = o.recurseForDerivations or true; }
+      )
+      m
```
Comment (domenkozar) [inline nix/dimension.nix]:
```suggestion
      attrs
```

**Review by domenkozar — CHANGES_REQUESTED** [review]

File: nix/ci.nix (hunk 1 above)
Comment (roberth) [reply nix/ci.nix]: bump'n'dump.

I guess branches will suffice indeed.

**Pull request merged**